Cleaning up a bit dts scripts

tbursztyka opened the pull request:
@b0661 could you take a look as well? Actually, I'd like to backport some of your patches around that tool as well (global stuff for modularisation, and the DTDirective class), as I'll need it to generate the spi-devices nodes.

@erwango maybe your patch about convert inherited 'id' to 'gen_type'  could be backported as well?

erwango committed:
scripts: extract_dts_include: convert inherited 'id' to 'node_type'

During yaml collapse step, convert inherited 'id' key to 'node_type'.
With this new 'node_type', it's is more easy to apply common treatment
to all bindings that include the same base yaml file but might not
have similar bindings/constraint naming convention.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

b0661 committed:
scripts: extract_dts_includes.py: factorize globals for module usage

Prepare to split extract_dts_includes in modules.

extract_dts_includes design is based on globals. Every module must be
able to use (import) them.

Signed-off-by: Bobby Noelte <b0661n0e17e@gmail.com>

Commit:
scripts/dts: Make extract_dts_includes generating both files by itself

Thus no need to parse twice the .dtsi, .dts and .yaml files.
Simplify provided arguments relevancy, letting argparse generating usage
message by itself.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
scripts/dts: De-clutter main function in extract_dts_includes

Moving logical parts into dedicated functions for clarity.
Factorizing a bit the code for potential future changes (more
'zephyr,xxxx' directives related to CONFIG_ option).

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
scripts/dts: Fix Copyright headers

One was missing it and the other was not the new format.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
scripts/dts: Use 4-spaces tabs instead of 2-space tabs in devicetree.py

Mandated by Python PEP-8.
(And normalize the way we write python in dts scripts also)

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

galak reviewed (dismissed):
Can you put the cleanup removal of 'return' into its own commit

Commit:
script/dts: Remove unnecessary empty return on functions

No need to use return at the end of a function when it's not returning
anything.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
hum I messed up something, will see that later today

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
+                fd.write("%s=%s\n" % (prop, defs[node].get(prop)))
 
-        sys.stdout.write("\n")
+        fd.write("\n")
```
Don't you have to close fd?

Perhaps you could use "with" instead of "try".

tbursztyka replied:
it's run-once script and the files are opened only once, since all the fd are closed on exit I guess it's ok not to specifically close them.

SebastianBoe replied:
It's just good practice though, if I see this when reading code I have to read the rest of the program to verify that nothing else is opening the file at the same time.

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
+def generate_keyvalue_file(defs, kv_file):
+    try:
+        fd = open(kv_file, "w")
+    except:
```
One should filter on the exception type that is being expected.

Also, I'm not sure if the new Exception actually is an improvement on what open() would have thrown.

```
>>> open("some_file.txt", "w")
<open file 'some_file.txt', mode 'w' at 0x7f6925307540>
>>> open("a/path/to/some_file.txt", "w")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 2] No such file or directory: 'a/path/to/some_file.txt'

```

tbursztyka replied:
I copy/pasted what was done with the dts file, but indeed better let the system's exception going instead of a custom one.

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
-def generate_include_file(defs, args):
+def generate_include_file(defs, inc_file, fixups):
+    try:
+        fd = open(inc_file, "w")
```
An earlier review comment applies here as well.

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
+def generate_include_file(defs, inc_file, fixups):
+    try:
+        fd = open(inc_file, "w")
+    except:
```
An earlier review comment applies here as well.

b0661 commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -849,69 +840,93 @@ def main():
+    get_compatibles(dts['/'], '/')
+    get_phandles(dts['/'], '/', {})
+    get_aliases(dts['/'])
+    get_chosen(dts['/'])
```
The useful lists are all views on the same DTS. Maybe they can be encapsulated in a DTS class with access methods. That way this class may become an extra module.

tbursztyka replied:
Indeed, that could enable moving towards a full class-based tool along with your DTDirective class  design.

b0661 commented:
@tbursztyka 
you may apply https://github.com/zephyrproject-rtos/zephyr/pull/5799/commits/20eb6f400901842ffa51a2ccf4bceaaeb4226860 to prepare for the restructuring and de-clutter of extract_dts_includes.

erwango commented:
> @erwango maybe your patch about convert inherited 'id' to 'gen_type' could be backported as well?

Sure! Looking at it again, we might review name 'gen_type' to something more explicit.
But I think this is indeed useful.

erwango commented:
> we might review name 'gen_type' to something more explicit.

Here is a branch on my tree with new name:
https://github.com/erwango/zephyr/tree/dts_node_type

galak commented:
@SebastianBoe you good with this?

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -667,51 +520,56 @@ def print_key_value(k, v, tabstop):
+
+        fd.write("\n")
+
+    fd.close()
```
I'm sorry, this is not sufficient.

From PEP8:

> When a resource is local to a particular section of code, use a with statement to ensure it is cleaned up promptly and reliably after use. A try/finally statement is also acceptable.

SebastianBoe commented on scripts/dts/extract_dts_includes.py:
```diff
@@ -732,78 +590,42 @@ def generate_include_file(defs, args):
+def load_and_parse_dts(dts_file):
+    with open(dts_file, "r") as fd:
+        dts = parse_file(fd)
+        fd.close()
```
The 'with' statement will close fd, so you don't have to explicitly close it here. This close() function call must be removed to not confuse the reader.

SebastianBoe commented:
"@SebastianBoe you good with this?"
Not yet, I have reviewed the review-fixes now.

tbursztyka commented:
@galak is it all fine now?

nashif reviewed (approved):
LGTM
